tests/unit/test_oaipmh.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2017 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+from datetime import datetime
+import json
+from mock import patch
+from os import remove, rmdir
+import pytest
+
+from hepcrawl.spiders.oaipmh_spider import OAIPMHSpider, _Granularity
+from scrapy.utils.project import get_project_settings
+
+
+def override_dynamic_fields(run):
+    if 'last_run_finished_at' in run:
+        run['last_run_finished_at'] = '2017-12-08T23:55:54.794969'
+    return run
+
+
+@pytest.fixture(scope='function')
+def cleanup():
+    yield
+    remove('/tmp/last_runs/OAI-PMH/2cea86bbc1d329b4273a29dc603fb8c0bb91439c.json')
+    rmdir('/tmp/last_runs/OAI-PMH')
+    rmdir('/tmp/last_runs')
+
+
+@pytest.fixture
+def settings():
+    settings_patch = {
+        'LAST_RUNS_PATH': '/tmp/last_runs/'
+    }
+    settings = get_project_settings()
+    with patch.dict(settings, settings_patch):
+        yield settings
+
+
+@pytest.fixture
+def spider(settings):
+    spider = OAIPMHSpider('http://export.arxiv.org/oai2', settings=settings)
+    spider.from_date = '2017-12-08'
+    spider.set = 'physics:hep-th'
+    spider.metadata_prefix = 'marcxml'
+    yield spider
+
+
+def test_last_run_file_path(spider):
+    expected = '/tmp/last_runs/OAI-PMH/2cea86bbc1d329b4273a29dc603fb8c0bb91439c.json'
+    result = spider._last_run_file_path()
+    assert expected == result
+
+
+def test_store_and_load_last_run(spider, cleanup):
```

Comment:
name of the tests (by convention) should be `test_<thing being tested>_<test case details>`